core/src/main/java/com/graphhopper/routing/template/ViaRoutingTemplate.java

```diff
@@ -65,6 +66,9 @@ public ViaRoutingTemplate(GHRequest ghRequest, GHResponse ghRsp, LocationIndex l
         queryResults = new ArrayList<>(points.size());
         for (int placeIndex = 0; placeIndex < points.size(); placeIndex++) {
             GHPoint point = points.get(placeIndex);
+            if(ghRequest.hasPointHints()){
+                edgeFilter = new LevensteinDistanceEdgeFilter(encoder, ghRequest.getPointHint(placeIndex));
```

Comment:
I would use a wrapper of the old edgeFilter, otherwise we could accept stuff that edgeFilter would reject.